readthedocs/proxito/views/mixins.py

```diff
@@ -287,8 +287,15 @@ def get_redirect_response(self, request, redirect_path, proxito_path, http_statu
         """
         # `proxito_path` doesn't include query params.
         query = urlparse(request.get_full_path()).query
-        # Pass the query params from the original request to the redirect.
-        new_path = urlparse(redirect_path)._replace(query=query).geturl()
+
+        # Combine the query params from the original request with the ones from the redirect.
+        redirect_parsed = urlparse(redirect_path)
+        redirect_query = redirect_parsed.query
+        if redirect_query:
+            if query:
+                query += "&"
```

Comment:
Shouldn't we use a Python function to concatenate these and encode the query strings properly? https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs and https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode

I think it's safer to use these functions and do it manually.